EpikCord/file.py

```diff
@@ -0,0 +1,37 @@
+import mimetypes
+import io
+from typing import Optional
+
+class File:
+    def __init__(
+        self,
+        filename: str,
+        contents: io.IOBase,
+        *,
+        description: Optional[str] = None,
+        mime_type: Optional[str] = None,
+        spoiler: bool = False,
+    ):
+        """
+        Parameters
+        ----------
+        filename: str
+            The filename of the file.
+        contents: io.IOBase
+            The contents of the file.
+        mime_type: Optional[str]
+            The mime type of the file. If not provided, it will be guessed.
+        spoiler: bool
+            Whether or not the file is a spoiler.
+        ephemeral: bool
+            Whether or not the file is ephemeral.
+        description: Optional[str]
+            The description of the file.
+        """
+        self.contents: io.IOBase = contents
+        self.mime_type: Optional[str] = mime_type or mimetypes.guess_type(filename)[0]
+        if spoiler:
+            self.filename: str = f"SPOILER_{filename}"
+        else:
+            self.filename: str = filename
```

Comment:
Could be a property like `spoiler`?

Reply:
Introducing a small hierarchy for the `File` class would be nice

Reply:
This is how Discord handles images that are spoilers, they are prefixes with `SPOILER_`, having a property is useless.

Reply:
What is meant by a "small hierarchy"?

Reply:
> What is meant by a "small hierarchy"?

Split the class into multiple